src/pages/form/advanced-form/components/TableForm.tsx

```diff
@@ -0,0 +1,297 @@
+              placeholder="工号"
+            />
+          );
+        }
+        return text;
+      },
+    },
+    {
+      title: '所属部门',
+      dataIndex: 'department',
+      key: 'department',
+      width: '40%',
+      render: (text: string, record: TableFormDateType) => {
+        if (record.editable) {
+          return (
+            <Input
+              value={text}
+              onChange={e => this.handleFieldChange(e, 'department', record.key)}
+              onKeyPress={e => this.handleKeyPress(e, record.key)}
+              placeholder="所属部门"
+            />
+          );
+        }
+        return text;
+      },
+    },
+    {
+      title: '操作',
+      key: 'action',
+      render: (text: string, record: TableFormDateType) => {
+        const { loading } = this.state;
+        if (!!record.editable && loading) {
+          return null;
+        }
+        if (record.editable) {
+          if (record.isNew) {
+            return (
+              <span>
+                <a onClick={e => this.saveRow(e, record.key)}>添加</a>
+                <Divider type="vertical" />
+                <Popconfirm title="是否要删除此行？" onConfirm={() => this.remove(record.key)}>
+                  <a>删除</a>
+                </Popconfirm>
+              </span>
+            );
+          }
+          return (
+            <span>
+              <a onClick={e => this.saveRow(e, record.key)}>保存</a>
+              <Divider type="vertical" />
+              <a onClick={e => this.cancel(e, record.key)}>取消</a>
+            </span>
+          );
+        }
+        return (
+          <span>
+            <a onClick={e => this.toggleEditable(e, record.key)}>编辑</a>
+            <Divider type="vertical" />
+            <Popconfirm title="是否要删除此行？" onConfirm={() => this.remove(record.key)}>
+              <a>删除</a>
+            </Popconfirm>
+          </span>
+        );
+      },
+    },
+  ];
+
+  constructor(props: TableFormProps) {
+    super(props);
+    this.state = {
+      data: props.value,
+      loading: false,
+      value: props.value,
+    };
+  }
+
+  getRowByKey(key: string, newData?: TableFormDateType[]) {
+    const { data = [] } = this.state;
+    return (newData || data).filter(item => item.key === key)[0];
+  }
+
+  toggleEditable = (e: React.MouseEvent | React.KeyboardEvent, key: string) => {
+    e.preventDefault();
+    const { data = [] } = this.state;
+    const newData = data.map(item => ({ ...item }));
+    const target = this.getRowByKey(key, newData);
+    if (target) {
+      // 进入编辑状态时保存原始数据
+      if (!target.editable) {
+        this.cacheOriginData[key] = { ...target };
+      }
+      target.editable = !target.editable;
+      this.setState({ data: newData });
+    }
+  };
+
+  newMember = () => {
+    const { data = [] } = this.state;
+    const newData = data.map(item => ({ ...item }));
+    newData.push({
+      key: `NEW_TEMP_ID_${this.index}`,
+      workId: '',
+      name: '',
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The key 'name' is not sorted alphabetically](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)